libs/cardano-ledger-core/src/Cardano/Ledger/UMapCompact.hs

```diff
@@ -98,15 +107,40 @@ import GHC.Stack (HasCallStack)
 import NoThunks.Class (NoThunks (..))
 import Prelude hiding (lookup)
 
+-- ================================================
+
+-- A Reward-Deposit Pair, will be used to represent the reward
+-- and the deposit for a given (Credential 'Staking c)
+data RDPair = RDPair
+  { rdReward :: {-# UNPACK #-} !(CompactForm Coin)
+  , rdDeposit :: {-# UNPACK #-} !(CompactForm Coin)
+  }
+  deriving (Show, Eq, Ord, Generic, NoThunks, NFData)
+
+rdPair :: Coin -> Coin -> RDPair
+rdPair rew dep = RDPair (compactCoinOrError rew) (compactCoinOrError dep)
+
```

Comment:
This function is dangerous and is not used anywhere. There is a similar one redefined in tests, but this one should be removed
```suggestion
```